core/src/main/resources/spring-rmapcore-context.xml

```diff
@@ -17,7 +17,9 @@
  	<bean id="discomgr" class="info.rmapproject.core.rmapservice.impl.openrdf.ORMapDiSCOMgr" scope="prototype" parent="abstractManager"/>
  	<bean id="statementmgr" class="info.rmapproject.core.rmapservice.impl.openrdf.ORMapStatementMgr" scope="prototype" parent="abstractManager"/>
  	<bean id="eventmgr" class="info.rmapproject.core.rmapservice.impl.openrdf.ORMapEventMgr" scope="prototype" parent="abstractManager"/>
- 	<bean id="agentmgr" class="info.rmapproject.core.rmapservice.impl.openrdf.ORMapAgentMgr" scope="prototype" parent="abstractManager"/>
+ 	<bean id="agentmgr" class="info.rmapproject.core.rmapservice.impl.openrdf.ORMapAgentMgr" parent="abstractManager">
+		<constructor-arg ref="eventmgr"/>
+	</bean>
```

Comment:
I don't think you need to use the constructor-arg if you use Autowired annotation.  Unless there is a reason to avoid autowired for this, might be good to stay consistent. Autowired used here: 
https://github.com/rmap-project/rmap/blob/master/core/src/main/java/info/rmapproject/core/rmapservice/impl/openrdf/ORMapDiSCOMgr.java
Noting also that you removed the scope=prototype property. Just wanted to check if this was deliberate. It was added in the course of troubleshooting some issues with IDs crossing over threads. The default scope is singleton if prototype is not defined. (https://docs.spring.io/spring/docs/3.0.0.M3/reference/html/ch04s04.html).

Reply:
The usage of annotations vs XML is quite inconsistent throughout the codebase.  Consistency is good.  There will be places where XML will have to be used, though (while annotations gets you a lot of the way from removing XML, you can't get totally away from it).

Regarding scope, I'm not sure that was deliberate, to be honest I don't remember.  I don't like the use of prototype beans, in general, unless there's a reason for it.  So I would be for replacing the prototype scope with singleton beans unless there's a good reason for using prototype.  That's probably a different PR.

Reply:
When we did load testing there was a problem with not being able to retrieve many of the DiSCOs that we had written. ID assignments were crossing over threads. This was part of a sledge hammer approach to preventing that - albeit a potentially misguided one!  Since then I haven't seen that problem again, so I'm a bit concerned about changing anything without completing issue #61 and verifying that we won't run into problems after the changes.  If you're confident this won't affect it then I'll merge.